ckanext/ssm/plugin.py

```diff
@@ -28,16 +37,20 @@ def get_instance_identity_document():
     ).json()
 
 
-class SSMConfigPlugin(SingletonPlugin):
+class SSMConfigPlugin(plugins.SingletonPlugin):
     """Interpolate SSM Parameter Store values into config.
     """
-    implements(IConfigurer, inherit=True)
+    plugins.implements(plugins.IConfigurer, inherit=True)
+
+    client: 'SSMClient|None' = None
```

Comment:
It would probably be simpler to pass this to the functions as a parameter. We can do the null check in the `update_config` function and then guarantee it's not null when passing it on.